core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1753,9 +1766,21 @@ class DAGScheduler(
     messageScheduler.shutdownNow()
     eventProcessLoop.stop()
     taskScheduler.stop()
+    heartbeater.stop()
+  }
+
+  /** Reports heartbeat metrics for the driver. */
+  private def reportHeartBeat(): Unit = {
```

Comment:
With cluster mode, including YARN, there isn't a local executor, so the metrics for the driver would not be collected. Perhaps this could be modified to skip this step for local mode.

Reply:
> With cluster mode, including YARN, there isn't a local executor, so the metrics for the driver would not be collected.

Yes. But the problem is can we use `executor`'s  `getCurrentExecutorMetrics()` method for collecting memory metrics for `driver` ? IIRC, `driver`  do not acqurie memory from execution memory pool at least.

Reply:
It's a bit redundant for fields that aren't used by the driver -- for the driver, execution memory gets set to 0.